processor/pom.xml

```diff
@@ -145,6 +149,11 @@
                         </goals>
                         <configuration>
                             <minimizeJar>true</minimizeJar>
+                            <artifactSet>
+                                <excludes>
+                                    <exclude>${project.groupId}:mapstruct-spi</exclude>
```

Comment:
IMO, we should include the SPI parts into the fat-jar as well (without shading into a different package, of course), as they are always required at runtime of the processor.

When adding the processor manually to javac or in an IDE (e.g. Eclipse), you currently only need to add one mapstruct-processor.jar - I think that is pretty convinient and it would be great if we could keep it that way. 

WDYT?